src/App.tsx

```diff
@@ -1,61 +1,133 @@
 import './App.scss';
+import { useState } from 'react';
 
-// import usersFromServer from './api/users';
-// import todosFromServer from './api/todos';
+import { TodoList } from './components/TodoList';
+
+import usersFromServer from './api/users';
+import todosFromServer from './api/todos';
+import { TodoWithUser } from './types/Todo';
+
+const todosWithUsers: TodoWithUser[] = todosFromServer.map(todo => {
+  const user =
+    usersFromServer.find(person => person.id === todo.userId) || null;
+
+  return {
+    ...todo,
+    user,
+  };
+});
+
+const findMaxId = (todos: TodoWithUser[]) => {
+  const ids = todos.map(todo => todo.id);
+
+  return Math.max(...ids);
+};
 
 export const App = () => {
+  const [todos, setTodos] = useState<TodoWithUser[]>(todosWithUsers);
+
+  const [title, setTitle] = useState('');
+  const [titleError, setTitleError] = useState(false);
+
+  const [selectedUserId, setSelectedUserId] = useState('0');
+  const [selectedUserIdError, setSelectedUserIdError] = useState(false);
+
+  const handleResetForm = () => {
+    setTitle('');
+    setTitleError(false);
+
+    setSelectedUserId('0');
+    setSelectedUserIdError(false);
+  };
+
+  const handleAddTodo = (event: React.FormEvent<HTMLFormElement>) => {
+    event.preventDefault();
+
+    const isTitleValid = !!title.trim();
+    const isUserValid = selectedUserId !== '0';
+
+    if (!isTitleValid) {
+      setTitleError(true);
+    }
+
+    if (!isUserValid) {
+      setSelectedUserIdError(true);
+    }
+
+    if (!isTitleValid || !isUserValid) {
+      return;
+    }
+
+    const newTodo: TodoWithUser = {
+      id: findMaxId(todos) + 1,
+      title,
+      userId: +selectedUserId,
+      completed: false,
+      user:
+        usersFromServer.find(person => person.id === +selectedUserId) || null,
+    };
+
+    setTodos(current => [...current, newTodo]);
+
+    handleResetForm();
+  };
+
   return (
     <div className="App">
       <h1>Add todo form</h1>
 
-      <form action="/api/todos" method="POST">
+      <form action="/api/todos" method="POST" onSubmit={handleAddTodo}>
```

Comment:
It's better to move all parameters onto separate lines, because there are more than two of them, and it's hard to read them.